### DISCUSSION TITLE
Color bar in all subplots

### DISCUSSION THREAD

**Discussion by ldv1** [pyvista/pyvista#3476] [Q&A]
Hi,

I made a figure with subplots and would like to print the color bar in each subplot.
Surprisingly, the color bar appears on the first subplot but not on the others. Furthermore, the color of the object (a sphere) is wrong.
Here is my code:

```
import pyvista as pv

sphere = pv.Sphere()
n = sphere.n_points

plotter = pv.Plotter(shape=(1,4), window_size=[1000,400])
for i in range(4):
    signal = [i]*n
    plotter.subplot(0,i)
    plotter.add_mesh(sphere, scalars=signal, show_scalar_bar=True, clim=[0,3])

plotter.link_views()
plotter.show()
```

I get

![image](https://user-images.githubusercontent.com/12841688/195985497-37b7e966-c5ff-4533-b5e3-8ed82dc9d758.png)

I expect each sphere to have a distinct color because I use `clim = [0,3]` for all of them.

Thanks for the help.

**Comment by ldv1** [comment]
I can answer part of my own question: to get the true colors, add `copy_mesh=True` to `plotter.add_mesh(...)`, then I get

![image](https://user-images.githubusercontent.com/12841688/196431769-e82425b9-7fbe-4a12-8851-ba86aca1d1d5.png)

This is not intuitive because the mesh is the same, it is only the scalar fields I pass that are different.

There only remains the issue with the color bar that only appears on the first subplot, but not on the others. I hope someone can post a solution.

**Comment by MatthewFlamm** [comment]
I think the issue is that when you explicitly pass in the scalars as a number to `add_mesh` it implicitly stores it as the same scalar.  The Plotter thinks that you are actually plotting the same scalar, ignoring your selection for `show_scalar_bar`.  This seems like a bug to me, but I'm not sure if there is some structural vtk reason this cannot be accomplished.

Here is the way to fix your issue, ~which also reduces the memory footprint of your code since it doesn't copy the mesh itself~.  Edit: in fact this increases the memory footprint, which is not ideal.  You can store all of your data under a different key name on the same mesh.  PS you can get the same behavior of your code by using the commented out version of this code, which confirms that it is a case of using the same name for all the scalars.

```py

import pyvista as pv

sphere = pv.Sphere()
n = sphere.n_points
for i in range(4):
    sphere[f"{i}"] = [i] * n
    # sphere["1"] = [i] * n
plotter = pv.Plotter(shape=(1,4), window_size=[1000,400])
for i in range(4):
    plotter.subplot(0,i)
    plotter.add_mesh(sphere, scalars=f"{i}", show_scalar_bar=True, clim=[0,3], copy_mesh=True)
    # plotter.add_mesh(sphere, scalars="1", show_scalar_bar=True, clim=[0,3], copy_mesh=True)

plotter.link_views()
plotter.show()
```

![image](https://user-images.githubusercontent.com/39341281/196436636-00665d90-2e7b-4db4-aca4-800e4f83193f.png)
